src/changelog-check.ts

```diff
@@ -0,0 +1,200 @@
+  } catch (error) {
+    console.error('❌ Error parsing CHANGELOG.md:', error);
+    return '';
+  }
+}
+
+/**
+ * Displays the differences between two sets of changelog entries.
+ * @param baseChanges - The content of the '[Unreleased]' section from the base branch's CHANGELOG.md.
+ * @param featureChanges - The content of the '[Unreleased]' section from the feature branch's CHANGELOG.md.
+ */
+function displayDiff(baseChanges: string, featureChanges: string) {
+  // Compute the line-by-line differences
+  const differences = diffLines(baseChanges, featureChanges);
+
+  const addedLines: string[] = [];
+  const removedLines: string[] = [];
+
+  // Collect added and removed lines into separate lists
+  differences.forEach((part) => {
+    if (part.added) {
+      addedLines.push(part.value.trim()); // Trim to remove leading/trailing spaces
+    } else if (part.removed) {
+      removedLines.push(part.value.trim());
+    }
+  });
+
+  // Print the diff summary
+  console.log("🔍 Diff between base and feature '[Unreleased]' sections:");
+
+  if (removedLines.length > 0) {
+    console.log('❌ Removed:');
+    removedLines.forEach((line) => console.log(`${line}`));
+  } else {
+    console.log('❌ No removed lines.');
+  }
+
+  if (addedLines.length > 0) {
+    console.log('✅ Added:');
+    addedLines.forEach((line) => console.log(`${line}`));
+  } else {
+    console.log('✅ No added lines.');
+  }
+}
+
+/**
+ * Validates that the CHANGELOG.md in a feature branch has been updated correctly by comparing it
+ * against the CHANGELOG.md in the base branch.
+ * @param repo - The GitHub repository from which to fetch the CHANGELOG.md file.
+ * @param baseBranch - The base branch (typically 'main' or 'master') to compare against.
+ * @param featureBranch - The feature branch that should contain the updated CHANGELOG.md.
+ */
+async function validateChangelog(
+  repo: string,
+  baseBranch: string,
+  featureBranch: string,
+) {
+  console.log(`🔍 Fetching CHANGELOG.md from GitHub repository: ${repo}`);
+
+  // Fetch CHANGELOG.md from both branches
+  const baseChangelogContent = await fetchChangelogFromGitHub(repo, baseBranch);
+  const featureChangelogContent = await fetchChangelogFromGitHub(
+    repo,
+    featureBranch,
+  );
+
+  if (!featureChangelogContent) {
+    console.error('❌ CHANGELOG.md is missing in the feature branch.');
+    throw new Error('❌ CHANGELOG.md is missing in the feature branch.');
+  }
+
+  // Parse the changelogs
+  const baseChanges = await parseChangelog(baseChangelogContent);
+  const featureChanges = await parseChangelog(featureChangelogContent);
+
+  console.log('🔍 Comparing changelog entries...');
+
+  console.log('Base unreleased section:', baseChanges);
+  console.log('Feature unreleased section:', featureChanges);
+
+  displayDiff(baseChanges, featureChanges);
+
+  if (baseChanges === featureChanges) {
+    console.log(
+      "❌ No new entries detected under '## Unreleased'. Please update the changelog.",
```

Comment:
Two suggestions related to this:
* This suggests that updating the changelog is the recommended way to address this error, but that's not necessarily true. We really don't want junk entries for non user-facing changes, that would make releases much more difficult rather than easier. We should be suggesting that the contributor either document user-facing changes, or add `no-changelog` to bypass the check if there are no user-facing changes.
* Have you considered putting this in a comment rather than the console output? It will be much easier to see this feedback if it's in a comment, particularly as we first roll out this workflow.